File: CentralAuth/Commons/Models/UserProject.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace CentralAuth.Commons.Models
{
    public class UserProject : MetaClass
    {
        [Key]
        public int Id { get; set; }
        public string PenggunaNik { get; set; }
        public User Pengguna { get; set; }
        public string ProjekApiName { get; set; }
        public Project Projek { get; set; }
    }
}

File: CentralAuth/ClientApp/src/app/models/user-project.ts
import { User } from './user';
import { Project } from './project';

export class UserProject {
    id: number;
    penggunaNik: string;
    pengguna: User;
    projekApiName: string;
    projek: Project;
    public constructor(init?: Partial<UserProject>) {
        Object.assign(this, init);
    }
}
